bsdlib/README.rst

```diff
@@ -3,9 +3,131 @@
 BSD Library
 ###########
 
+The BSD library is Nordic Semiconductor's implementation of the BSD Socket API, which is a set of standard function calls that can be used in an application.
+This library aims to be compatible with the BSD socket standard to simplify writing and porting of existing applications. 
+
+The library has the following set of sockets:
+
+* 8 generic sockets that can be used generically between UDP, TCP, TLS, DTLS, and AT commands.
+* 1 GNSS/AGPS socket
+* 1 Modem DFU socket
+* 10 PDN sockets
+
+Overview
+********
+
+.. figure:: doc/images/bsd_lib_overview.svg
+   :alt: BSD library overview diagram
+
+   BSD library overview diagram
+
+The BSD library is the primary interface to operate the nRF9160 modem in order to establish LTE-M, NBIOT, and GNSS connections.
+The library abstracts away the details on memory management and synchronization leaving the application to focus on writing the application using standardized API calls.
+
+The current implementation of the BSD library supports the following protocols:
+
+* TCP (client/server)
+* UDP
+* RAW sockets
+* DTLS 1.2 client
+* TLS 1.2 client
+* AT commands
+* GNSS
+* AGPS
+* PDN management
+* Modem DFU
+
+.. note::
+   The DNS address management protocol is not yet implemented.
+
+The library implements the following standard BSD functions:
```

Comment:
While this is a complete list, each socket supports its own subset of this API e.g. the modem DFU socket doesn't support `listen` or `poll` and it doesn't make sense for it to support those either. Instead of mentioning that `select` is not yet implemented (line 67), we could either provide a more exhaustive list of which functions are supported/implemented on each socket, or just have a note saying that each socket supports a subset of this interface.

Reply:
Changing the note at the end of the list to:
"Each socket supports a subset of these functions."